lib/robot.js

```diff
@@ -25,6 +25,25 @@ class Robot {
     });
   }
 
+  addEventToListeners(event, callback) {
```

Comment:
@bkeepers The recursive solutions is beautiful! 😍  So much less code needed 💯 
I've implemented that one for now.